src/Misc/Part.cpp

```diff
@@ -165,8 +165,12 @@ static const Ports partPorts = {
 
             //d.broadcast("/damage", "s", part_loc);
         }},
-
-
+        {"savexml:", rProp(internal) rDoc("Save Part to the file it has been loaded from"), 0,
+        [](const char *, RtData &d)
+        {
+            Part *p = (Part*)d.obj;
+            p->saveXML(p->loaded_file);
```

Comment:
If `p->loaded_file` has never been used, I get `Conditional jump or move depends on uninitialised value(s)` from valgrind.

Reply:
> If `p->loaded_file` has never been used, I get `Conditional jump or move depends on uninitialised value(s)` from valgrind.

yes, it doesn't work with a new preset, that has never been saved. easiest way would be not saving at all in such a case. 

to prevent loss of creative work it could be saved to the first free slot in a bank reserved for this case.
slightly modified this could serve another use case i have thought of for a long time: an easy way to save the steps of your creative process.
In the old gui you could do this by using the write button on empty slots.

Reply:
@fundamental What's your opinion about this suggestion?

Reply:
New idea: if the part has never been loaded or saved, the savexml port handler makes up a new one from date and time.
still open question: what if i press save again after editing. new filename or overwrite the old file. I tend to overwrite only files that i have loaded or given a name myself.

Reply:
With regards to the suggestion, there's certainly some flexibility in the new design as not a lot of work has been put into making it friendly to save instruments to banks. Having one slot reserved as a defacto work in progress slot is interesting. Not sure if users would like it or not, but I'm not raising any immediate red flags. IMO the major thing to watch out for is creating a workflow which results in users unintentionally overwriting an old patch while expecting to save it elsewhere. Saving based upon date/time combinations is one approach.

For many users you have the difference of a bank that they control and a bank which they've received i.e. during normal operation a bank is either read only or it is editable by a user. Perhaps some way of expressing that would be good, though I'm unsure if this is too big of an issue to discuss given the current scope of this PR.

Reply:
Indeed I didnt think of the built-in-preset use case...

I think the read only could be accomplished by filesystem permissions.
-> if a read-only file is loaded, the save button should be disabled.

option 2: in linux the built-in vs user banks can be separated by using different directories /usr/ vs /home/. is there something similar on windows?
in built-in banks the save button should also be disabled.

Reply:
For me, load/save to/from disk and load/save to/from a bank are two different concepts. I never used banks in the past and I would not assume to find a not-saved file there. I would not like to learn the banks concept just because I used this functionality without prior loading of the part.

I'd rather do it like in most applications: Of you click "save as", of course, always a dialogue comes. If you click save and it has not yet been loaded/saved, again a dialogue, but if it has, direct saving. In this case, zyn send the GUI a boolean in return to say "I could not do a 'save'", and the UI can then raise a dialogue (or an error window).

Reply:
I work with banks because I use them in rehearsal room headless via midi only. if I notice that something needs to be altered, I write a note, change the bank back at home at my desktop and rsync it to the headless box.
My workflow is: load one preset from my bank (in browser or via PGM Change message),  change a few parameters unti I like it again and then save it to the same slot.